workflow-controller/controller-service/src/main/java/io/hyscale/controller/commands/HyscaleDeployServiceCommand.java

```diff
@@ -138,26 +141,54 @@ public void run() {
             // clean up service dir before dockerfileGen
             workflowContext.addAttribute(WorkflowConstants.CLEAN_UP_SERVICE_DIR, true);
 
-            // Generate Docker file
-            dockerfileGeneratorComponentInvoker.execute(workflowContext);
+            try {
+                // Generate Docker file
+                dockerfileGeneratorComponentInvoker.execute(workflowContext);
+            } catch(HyscaleException e) {
+                logger.error("Error while generating dockerfile for app: {}, service: {}", appName, serviceName, e);
+                isFailed = true;
+            }
 
             if (!workflowContext.isFailed()) {
-                imageBuildComponentInvoker.execute(workflowContext);
+                try {
+                    imageBuildComponentInvoker.execute(workflowContext);
+                } catch (HyscaleException e) {
+                    logger.error("Error while building image for app: {}, service: {}", appName, serviceName, e);
+                    isFailed = true;
```

Comment:
Can we eliminate the duplicate code here ?

Reply:
ComponentInvoker internally checks if the workflow context is failed or not , so this check can be eliminated here I guess